src/js/nominatim.js

```diff
@@ -111,7 +115,7 @@ export class Nominatim {
     ajax.url = document.location.protocol + provider.url;
     ajax.data = provider.params;
 
-    if (options.provider === C.providers.BING) {
+    if (provider.callbackName) {
```

Comment:
Enable JSONP if the provider defines a `callbackName` in the object returned by `getParameters`.